### PR TITLE
Introduce a location overview page

### LINKED ISSUE #4035: Introduce a location overview page
## Abstract

We need to introduce an overview page per location, in a similar fashion to the asset details page.

The [LocationIcon.vue](https://github.com/rotki/rotki/blob/develop/frontend/app/src/components/history/LocationIcon.vue) component needs to change to act as a link when it is not part of a dropdown or list.

The routes for each details page would follow the schema `/location/:name` for example the location for `kraken` would be `/location/kraken`.

Each location should try to provide an overview of the balances/events etc for the specific location.

For example, the location `kraken` should show balances, trades, asset movements, staking along with any manual balances and ledger actions that have the kraken as the location.

Some locations such as `blockchain` might require some extra work since at least should provide an overview of anything blockchain-related.

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #4035

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.
- [ ] Introduuce a location overview page

### CODE REVIEW

**Review by kelsos — COMMENTED** [review]
Would it be easy to hide empty tables that are not loading?

File: frontend/app/src/store/balances/getters.ts [hunk 1]
```diff
@@ -947,29 +948,113 @@ export const getters: Getters<
...
       byLocation[location] = usdValue;
     }
 
+    const mainCurrency = session?.generalSettings.mainCurrency.tickerSymbol;
+    assert(mainCurrency, 'main currency was not properly set');
+
+    const currentExchangeRate = exchangeRate(mainCurrency);
+    const blockchainTotalConverted = currentExchangeRate
+      ? blockchainTotal.multipliedBy(currentExchangeRate)
+      : blockchainTotal;
```
Comment (kelsos) [inline frontend/app/src/store/balances/getters.ts:1038]: So this check will work when there is no exchange rate for the selected currency, but which cases are these specifically? ETH/BTC?

**Comment by kelsos** [comment]
@lukicenturi along with the fixing of the lint issues please add a change log entry.

Other than that if possible as I mentioned above could we hide any empty tables that are not loading, and also the question remains, about which currencies under what conditions do not return an exchange rate.

Other than that the PR is good.

File: frontend/app/src/store/balances/getters.ts (hunk 1 above)
Comment (lukicenturi) [reply frontend/app/src/store/balances/getters.ts:1038]: I'm not sure though. I saw that the currentExchangeRate can be `undefined` at `manualBalanceByLocation` getter, so I thought it is common to check it.

**Comment by lukicenturi** [comment]
> @lukicenturi along with the fixing of the lint issues please add a change log entry.
> 
> Other than that, if possible could we hide any empty tables that are not loading, and also the question remains, about which currencies under what conditions do not return an exchange rate.
> 
> Other than that the PR is good.

Changelog added, the empty table are hidden.

File: frontend/app/src/store/balances/getters.ts (hunk 1 above)
Comment (kelsos) [reply frontend/app/src/store/balances/getters.ts:1038]: no keep it, but I don't remember why the check is on manual balance, the only point I can think of this possibly being undefined is during the dashboard loading, but I am under the impression that we load the exchange rates early during the login.

**Review by kelsos — APPROVED** [review]
lgtm, thank you

**Pull request merged**